packages/cli-main/src/cli/commands/upgrade.ts

```diff
@@ -21,43 +22,27 @@ export default class Upgrade extends Command {
   }
 
   async run(): Promise<void> {
```

Comment:
I'd advise extracting all the business logic into a service to allow unit-testing instead of integration-testing through oclif's primitives.